lab-carolina/test/test.js

```diff
@@ -0,0 +1,24 @@
+//test require server.js and net module.
+
+const net = require('net');
+const server = require('../server.js');
+const assert = require('assert');
+
+describe('Testing server.js', function() {
+
+  describe('connection to the server', function() {
+    before(function(done) {
```

Comment:
This before block should be one level higher in scope. We want the server started before running any of the method level describe blocks. Inside each of those method level describe blocks though we can use another before to instantiate clients.